plugins/modules/database/saphana/hana_query.py

```diff
@@ -0,0 +1,197 @@
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: hana_query
+
+short_description: Execute SQL on HANA.
+
+version_added: "3.2.0"
+
+description: This module executes SQL statements on HANA with hdbsql.
+
+options:
+    sid:
+        description: The system ID.
+        type: str
+        required: true
+    instance:
+        description: The instance number.
+        type: str
+        required: true
+    user:
+        description: A dedicated username. Defaults to C(SYSTEM).
+        type: str
+        default: SYSTEM
+    password:
+        description: The password to connect to the database.
+        type: str
+        required: true
+    autocommit:
+        description: Autocommit the statement.
+        type: bool
+        default: true
+    host:
+        description: The Host IP address. The port can be defined as well.
+        type: str
+    database:
+        description: Define the database on which to connect.
+        type: str
+    encrypted:
+        description: Use encrypted connection. Defaults to C(false).
+        type: bool
+        default: false
+    filepath:
+        description:
+        - One or more files each containing one SQL query to run.
+        - Must be a string or list containing strings.
+        type: list
+        elements: path
+    query:
+        description:
+        - SQL query to run.
+        - Must be a string or list containing strings. Please note that if you supply a string, it will be split by commas (C(,)) to a list.
+          It's better to supply a one-element list instead to avoid mangled input.
+        type: list
+        elements: str
+
+author:
+    - Rainer Leber (@rainerleber)
+'''
+
+EXAMPLES = r'''
+
+- name: Simple select query
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    query: "select user_name from users"
+
+- name: Run several queries
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    query:
+    - "select user_name from users;"
+    - select * from SYSTEM;
+    host: "localhost"
+    autocommit: False
+
+- name: Run several queries from file
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    filepath:
+    - /tmp/HANA_CPU_UtilizationPerCore_2.00.020+.txt
+    - /tmp/HANA.txt
+    host: "localhost"
+
+
+'''
+
+RETURN = r'''
+
+query_result:
+    description: A JSON string of the returned values from the SQL queries.
+    returned: on success
+    type: list
+    elements: list
+    sample: [{"Column": "Value1"},{"Column": "Value2"}]
+
+
+'''
+
+import csv
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six import StringIO
+
+
+def csv_to_list(rawcsv):
+    lines = rawcsv[:rawcsv.rfind('\n')]
+    reader_raw = csv.DictReader(StringIO(lines))
+    reader = [dict((k, v.strip()) for k, v in row.items()) for row in reader_raw]
```

Comment:
Is the `strip()` really needed?

Reply:
unfortunatly yes, sometimes returned values have white spaces like "        'value test'     "

Reply:
But isn't that usually because spaces are part of the values in the database? In that case, stripping them would be returning wrong data.

Reply:
I don't think so without the strip it will look like this
```
ok: [10.2.0.16] => {
    "consistency_exec.query_result": [
        [
            {
                "AVG_TIME_S": "      0.99",
                "CHECK_ACTION": "CHECK",
                "CHECK_PROCEDURE_NAME": "CHECK_TABLE_CONSISTENCY",
                "DB_USER": "SYSTEM",
                "ERROR_DETAILS": "0",
                "EXECUTIONS": "         9",
                "LAST_START_TIME": "2021/05/24 21:24:36",
                "SCHEMA_NAME": "",
                "TABLE_NAME": "",
                "TOT_TIME_H": "      0.00"
            }
        ]
    ]
}
```